core/src/main/scala/kafka/utils/DelayedItem.scala

```diff
@@ -21,24 +21,19 @@ import java.util.concurrent._
 
 import org.apache.kafka.common.utils.Time
 
-import scala.math._
+class DelayedItem(val delayMs: Long, val time: Time) extends Logging {
+  def this(delayMs: Long) = this(delayMs, Time.SYSTEM)
```

Comment:
Do we need this? Can we not pass the `time` from every caller?

Reply:
We can, I was just trying to avoid the replica fetcher having to pass Time.SYSTEM in everywhere. If you prefer that I can change it. Is there any downside what I did though?

Reply:
It should pass the actual `time` instance so that we can mock if we want:

```class ReplicaFetcherThread(name: String,
                           fetcherId: Int,
                           sourceBroker: BrokerEndPoint,
                           brokerConfig: KafkaConfig,
                           failedPartitions: FailedPartitions,
                           replicaMgr: ReplicaManager,
                           metrics: Metrics,
                           time: Time,
```

Reply:
Thanks, yes, I was just coming back to say your suggestion was better for testability reasons.